Dynamic chart documentation (#82)
eProsima/Fast-DDS-monitor: docs/rst/getting_started/tutorial.rst

.. figure:: /rst/figures/screenshots/usage_example/init_domain.png
    :align: center

Add physical and logical panels
===============================

> By default, it is only displayed the DDS panel with where DDS entities are shown.

Reviewer: ```suggestion
By default, the monitor only display the DDS panel which lists the DDS entities together with their configuration and available statistics information.
```

Author: :+1:

Now, you have a *DomainParticipant* called :code:`Participant_sub`, that holds a *DataReader* called
:code:`DataReader_HelloWorldTopic_0.0.1.4`.
This *DataReader* has a locator, which will be the *Shared Memory Transport* locator.

> That is because the Monitor and the *DomainParticipant* are running in the same host, and so they use *shared memory*
> communication.

Reviewer: ```suggestion
That is because the Monitor and the *DomainParticipant* are running in the same host, and so they communicate using the Shared Memory Transport (SHM) protocol.
```

Author: :+1:

===================

In order to make the user experience easier, there is a possibility to change the name of an specific entity.

> We are going to change the name of our *Host* to make it simpler.

Reviewer: ```suggestion
We are going to change the name of our *Host* to make it easier to identify.
```

Author: :+1:

> For that, just do right click over the entity name and a dialog will open.

Reviewer: ```suggestion
For that, just right-click over the entity name and a menu with the available options for that entity will popup. Click on *Change alias* to re-name the entity.
```

Author: :+1:

.. figure:: /rst/figures/screenshots/usage_example/data_count_chart.png
    :align: center

.. todo::

    Comment depending on the screenshot

> 

Reviewer: Is this still a ToDo?

Author: :+1:

The statistic kinds that we are going to use for this example are:

* :code:`MEDIAN` (blue series)
* :code:`MAX`  (green series)
* :code:`MIN` (yellow series)
* :code:`STANDARD_DEVIATION` (purple series)

.. figure:: /rst/figures/screenshots/usage_example/latency_chart.png
    :align: center

It is worth mentioning that the series name, its color, the axis, and some features of the chart box could be changed
as mention in :ref:`chartbox`.

Create Dynamic Series Chart
===========================

This section describes how to graphically represent data in real-time of a running DDS network.

Periodic Latency Plot
---------------------

> This section explains how to represent the data regarding the latency in real-time between the publisher and

Reviewer: ```suggestion
This section explains how to represent the Fast DDS latency in real-time between the publisher and
```

Author: :+1:

the subscriber.
First of all, click in :ref:`display_dynamic_data_button`.
This will open a Dialog where you should choose one of the topics in which you want to see the data collected.

> For this case, we will choose :code:`FASTDDS_LATENCY`.

Reviewer: ```suggestion
For this case, choose :code:`FASTDDS_LATENCY`.
```

Author: :+1:

> We are going to set a :code:`Time window` of 1 minute.

Reviewer: ```suggestion
Set a :code:`Time window` of 1 minute.
```

Author: :+1:

> This means we will see the data of the last minute of the network.

Reviewer: ```suggestion
This means you will be able to see the data of the last minute of the network.
```

> We are going to set an :code:`Update period` of 5 seconds.

Reviewer: ```suggestion
Finally, set an :code:`Update period` of 5 seconds.
```

Author: :+1:

.. figure:: /rst/figures/screenshots/usage_example/new_dynamic_series_latency.png
    :align: center

After this, a new Dialog will open asking to configure the series that is going to be displayed.
In the case of :code:`FASTDDS_LATENCY` the data to show is related with two entities.
In our example we are going choose our *Host*.
This will retrieve the latency measured in the communication between the entities of this host to itself.
For this case, it is going to be the latency between the two participants, but this trick is very useful when
you want to filter latency between two specific hosts or even to collect all the latency in the same domain.

Now for the :code:`Statistics kind` option, we are going to use some of them in order to see more than one series of
statistical data.
Change the :code:`Statistics kind` and click :code:`Apply` for each of them in order to create a series for each one.
The statistic kinds that we are going to use for this example are:

* :code:`MEAN` (blue series)
* :code:`MAX` (green series)
* :code:`MIN` (yellow series)

.. figure:: /rst/figures/screenshots/usage_example/dynamic_latency_configuration.png
    :align: center

> This chart will be updated each 5 seconds, displaying the data collected within the last 5 seconds by the monitor.

Reviewer: ```suggestion
This chart will be updated each 5 seconds, displaying the data collected by the monitor within the last 5 seconds.
```

Author: :+1:

The axis are updated periodically, and so the zoom and chart move is not available in this kind of charts while
running.
For this propose, the *play/pause* button stops the axis to update, and so you could zoom and move along the chart.
Be aware that pausing the chart do not stop new points to appear, as every 5 seconds the update of the data will
still happen.

.. figure:: /rst/figures/screenshots/usage_example/dynamic_latency_chart.png
    :align: center

Latency DataPoints
------------------

There is a special feature for real-time data display that allow to see every *DataPoint* received from the DDS
entities monitored (similar to :code:`bins 0` in historic series).
In order to see this data in real-time, add a new series in this same chartbox in *Series->Add series*.
Choose again the *Host* as source and target and choose :code:`NONE` as :code:`Statistics kind`.

> Now we will see a new series (by default purple, in the example image red) that represents each of the *DataPoints*

Reviewer: ```suggestion
Now you should be able to see a new series (by default purple, in the example image red) that represents each of the *DataPoints*
```

Author: :+1:
